public/angular-app/configuration/config-controller.js

```diff
@@ -577,6 +589,7 @@ function ConfigController($routeParams, FilePathsFactory, ConfigFactory, Project
                     file['name'] = UtilityService.fileNameFromPath(file.centralPath);
                 });
                 vm.files = response.data;
+                vm.setFilesPage(vm.pageNum, vm.numItems);
```

Comment:
So this is not what I have imagined. You noticed how the whole thing is a little slow and hangs the UI. That's because it was rendering a dropdown with 3000+ items in it, but thats also because it was making a request to the DB and getting back 3000+ items. That's the culprit here. 

So have a look at how I setup the FilePaths table, or Warnings table. They are both using the Angular JS DataTables and they are both using server side processing. That has built in mechanisms for filtering and pagination, so that we don't have to implement it here like you did. It also returns back only items that are to be rendered in the table, and if more is needed it simply requests them from the DB. That's a little more chatter on DB side, but it's much more efficient on front end and rendering speed (unless DB is super busy and we are getting hangs that way, in which case we want to look into a load balancer like RabbitMQ and splitting up the DB into smaller chunks.)